ClientProxy.go

```diff
@@ -59,14 +65,22 @@ func (this ClientProxy) ServeDNS(w dns.ResponseWriter, request *dns.Msg) {
 		return
 	}
 	var requestBody []byte
+	//_, err = resp.Body.Read(requestBody)
 	nRead, err := resp.Body.Read(requestBody)
```

Comment:
The Go documentation uses a slightly diferent technique:

http://golang.org/pkg/net/http/

Perhaps something like this:

```go
defer resp.Body.Close()
body, err := ioutil.ReadAll(resp.Body)
```